Nexogen.Libraries.Metrics.Prometheus.Grpc/ClientMetricsInterceptor.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Threading.Tasks;
+using Grpc.Core;
+using Grpc.Core.Interceptors;
+
+namespace Nexogen.Libraries.Metrics.Prometheus.Grpc
+{
+    /// <summary>
+    /// gRPC client interceptor for collecting Prometheus metrics.
+    /// </summary>
+    public class ClientMetricsInterceptor : Interceptor
```

Comment:
You could put everything that is not externally consumed like ClientMetricsInterceptor  and ServerMetricsInterceptor (I don't know if I have missed something, I'm relying on the examples you have provided.) in namespace Nexogen.Libraries.Metrics.Prometheus.Grpc.Internal to signify that these are implementation details and not part of the stable API. This is what Microsoft is currently doing with their packages.

Reply:
I've moved `CountingStreamReader` and `CountingStreamWriter` to `Nexogen.Libraries.Metrics.Prometheus.Grpc.Internal`.